client/main.lua

```diff
@@ -51,24 +51,21 @@ end
 -- If in vehicle returns that, otherwise tries 3 different raycasts to get the vehicle they are facing.
 -- Raycasts picture: https://i.imgur.com/FRED0kV.png
 local function getVehicle()
-    local vehicle = cache.vehicle
     local raycastOffsetTable = {
         { fromOffset = vec3(0.0, 0.0, 0.0), toOffset = vec3(0.0, 20.0, -10.0) }, -- Waist to ground 45 degree angle
         { fromOffset = vec3(0.0, 0.0, 0.7), toOffset = vec3(0.0, 10.0, -10.0) }, -- Head to ground 30 degree angle
         { fromOffset = vec3(0.0, 0.0, 0.7), toOffset = vec3(0.0, 10.0, -20.0) }, -- Head to ground 15 degree angle
     }
 
-    local count = 0
-    while not vehicle and count < #raycastOffsetTable do
-        count += 1
-        vehicle = getVehicleInDirection(raycastOffsetTable[count]['fromOffset'], raycastOffsetTable[count]['toOffset'])
-    end
+    if not cache.vehicle then
+        for i = 1, #raycastOffsetTable do
+            local vehicle = getVehicleInDirection(raycastOffsetTable[i]['fromOffset'], raycastOffsetTable[i]['toOffset'])
 
-    if not IsEntityAVehicle(vehicle) then
-        vehicle = nil
+            if IsEntityAVehicle(vehicle) then
+                return vehicle
+            end
+        end
     end
-
-    return vehicle
 end
```

Comment:
I believe that we are missing the return case if no vehicle if this is called

Reply:
It is not necessary. The function returns nil if it does not find a vehicle.

Reply:
Gotcha. I didn't look to see where it all got called.